sale_order_field_advanced/models/sale_order_line_modification.py

```diff
@@ -0,0 +1,29 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, fields
+
+
+class SaleOrderLineModification(models.Model):
+    _name = 'sale.order.line.modification'
+    _description = 'Sale Order Line Modification'
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    price = fields.Monetary(currency_field='currency_id')
+    currency_id = fields.Many2one(
+        'res.currency',
+        string='Currency',
+        required=True,
+        default=lambda self: self.env.ref('base.EUR').id
+    )
```

Comment:
This is leak. We prefer this approach:
* Treat this new model as a multicompany model.
* Then, add a `company_id` field and add the required stuff with it (default value, record rule,...).
* When a modification is selected for a sale line, add check company stuff (like https://github.com/odoo/odoo/blob/3a28e5b0adbb36bdb1155a6854cdfbe4e7f9b187/addons/sale/models/sale_order_line.py#L244).
* At last, `currency_id` default value will be simply the company currency.

@ChristianSantamaria also check this with @lperez-solvos if you agree with me

Reply:
This is still a bit incomplete, we'll talk about it together @ChristianSantamaria @lperez-solvos